clang/lib/StaticAnalyzer/Checkers/PointerSubChecker.cpp

```diff
@@ -61,6 +61,10 @@ void PointerSubChecker::checkPreStmt(const BinaryOperator *B,
   if (LR->getSymbolicBase() || RR->getSymbolicBase())
     return;
 
+  if (!B->getLHS()->getType()->isPointerType() ||
+      !B->getRHS()->getType()->isPointerType())
+    return;
```

Comment:
Are sugared types passing this? Such as a typedef hiding the pointer type?
```c++
using ptr = int*;
```

Shouldnt we prefer canonical types instead?

Reply:
The function `isPointerType` uses already the canonical type.

Reply:
Awesome!